system/jhtree/jhtree.cpp

```diff
@@ -1957,6 +2048,12 @@ bool CKeyCursor::_ltEqual(IContextLogger *ctx)
             if (a==node->getNumKeys())
                 a--;   // value being looked for is off the end of the index.
             offset_t npos = node->getFPosAt(a);
+            if (depth < maxParentNodes)
+            {
+                parentNodeKeys[depth] = a;
+                parents[depth].swap(node);
+            }
+            lwm = 0;
```

Comment:
Is this a lurking bug?

Reply:
I don't think so, in that previously if lwm was non-zero we were already at leaf level and so would only go around the loop once. So you could argue it was a lurking bug but it's one that would never have been able to trigger previously.